lib/fileutils.rb

```diff
@@ -501,25 +501,81 @@ def cp_lr(src, dest, noop: nil, verbose: nil,
   end
   module_function :cp_lr
 
+  # Creates {symbolic links}[https://en.wikipedia.org/wiki/Symbolic_link].
   #
-  # :call-seq:
-  #   FileUtils.ln_s(target, link, force: nil, noop: nil, verbose: nil)
-  #   FileUtils.ln_s(target,  dir, force: nil, noop: nil, verbose: nil)
-  #   FileUtils.ln_s(targets, dir, force: nil, noop: nil, verbose: nil)
+  # When +src+ is the path to an existing file:
+  #
+  # - When +dest+ is the path to a non-existent file,
+  #   creates a symbolic link at +dest+ pointing to +src+:
+  #
+  #     FileUtils.touch('src0.txt')
+  #     File.exist?('dest0.txt')   # => false
+  #     FileUtils.ln_s('src0.txt', 'dest0.txt')
+  #     File.symlink?('dest0.txt') # => true
+  #
+  # - When +dest+ is the path to a directory,
+  #   creates a symbolic link at <tt>dest/src</tt> pointing to +src+:
+  #
+  #     FileUtils.touch('src1.txt')
+  #     FileUtils.mkdir('destdir1')
+  #     FileUtils.ln_s('src1.txt', 'destdir1')
+  #     File.symlink?('destdir1/src1.txt') # => true
+  #
+  # - When +dest+ is the path to an existing file,
+  #   creates a symbolic link at +dest+ pointing to +src+
+  #   if and only if keyword argument <tt>force: true</tt> is given
+  #   (raises an exception otherwise):
+  #
+  #     FileUtils.touch('src2.txt')
+  #     FileUtils.touch('dest2.txt')
+  #     FileUtils.ln_s('src2.txt', 'dest2.txt', force: true)
+  #     FileTest.symlink?('dest2.txt') # => true
+  #
+  #     FileUtils.ln_s('src2.txt', 'dest2.txt') # Raises Errno::EEXIST.
+  #
+  # When +src+ and +dest+ are both paths to directories,
+  # creates a symbolic link at <tt>dest/src</tt> pointing to +src+:
+  #
+  #   FileUtils.mkdir('srcdir3')
+  #   FileUtils.touch('srcdir3/src1.txt')
+  #   FileUtils.touch('srcdir3/src2.txt')
+  #   FileUtils.mkdir('destdir3')
+  #   FileUtils.ln_s('srcdir3', 'destdir3')
+  #   File.symlink?('destdir3/srcdir3') # => true
   #
-  # In the first form, creates a symbolic link +link+ which points to +target+.
-  # If +link+ already exists, raises Errno::EEXIST.
-  # But if the <tt>force</tt> option is set, overwrites +link+.
+  # When +src+ is an array of paths to existing files and +dest+ is a directory:
```

Comment:
Probably should mention what the actual behavior is in this case in the text, even if it can be inferred from the example.

Reply:
Added.